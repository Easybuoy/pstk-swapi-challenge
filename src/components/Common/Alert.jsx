import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';

import { Alert as StyledAlert } from '../../styles';

const Alert = ({ message }) => {
  const [hideMessage, setHideMessage] = useState(false);
  
  useEffect(() => {
    const timeout = setTimeout(() => {
      setHideMessage('none');
    }, 3000);
    return () => {
      clearTimeout(timeout);
    };
  }, []);

  return (
    <StyledAlert>
      <div className="toast" style={{ display: hideMessage }}>
        <p>{message}</p>
      </div>
    </StyledAlert>
  );
};

Alert.propTypes = {
  message: PropTypes.string.isRequired,
};

export default Alert;
