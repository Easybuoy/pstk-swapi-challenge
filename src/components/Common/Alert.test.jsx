import React from 'react';
import { shallow } from 'enzyme';

import Alert from './Alert';

describe('<Alert />', () => {
  const props = {
    message: 'Error'
  };

  it('renders the Alert component correctly', () => {
    const wrapper = shallow(<Alert {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
