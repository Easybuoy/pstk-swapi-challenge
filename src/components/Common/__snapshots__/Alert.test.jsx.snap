// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> renders the Alert component correctly 1`] = `
<styled.div>
  <div
    className="toast"
    style={
      Object {
        "display": false,
      }
    }
  >
    <p>
      Error
    </p>
  </div>
</styled.div>
`;
